cibyl/sources/elasticsearch/api.py

```diff
@@ -459,6 +464,26 @@ def get_tests(self, **kwargs):
                           )
                 continue
 
+            # If we filter by --test-duration then
+            # we don't accept empty time_duration
+            time_not_match = None
+            if test_duration_arguments:
+                # If we don't have time_duration
+                if not test_duration:
+                    continue
+                # For every condition passed by the user
+                for test_duration_argument in test_duration_arguments:
```

Comment:
This is a matter of preference but I would move this loop into a method that would return False if any condition is not met, and True otherwise. This way there is not need for `time_not_match` and the check in line 482

Reply:
It looks better. Thanks Joan!